doc/user/content/transform-data/idiomatic-materialize-sql/any.md

```diff
@@ -0,0 +1,173 @@
+---
+title: "`ANY()` equi-join condition"
+description: "Use idiomatic Materialize SQL for equi-join whose `ON` expression includes the `ANY()` operator."
+menu:
+  main:
+    parent: idiomatic-materialize-sql
+    identifier: idiomatic-materialize-any
+    weight: 5
+---
+
+## Overview
+
+The "`field = ANY(...)`" equality condition returns true if the equality
+comparison is true for any of the values in the `ANY()` expression.
+
+For equi-join whose `ON` expression includes an [`ANY` operator
+expression](/sql/functions/#expression-bool_op-any),
+Materialize provides an idiomatic SQL as an alternative to the `ANY()`
+expression.
+
+{{< callout >}}
+
+### Materialize and equi-join `ON fieldX = ANY(<array|list|map>)`
+
+When evaluating an equi-join whose `ON` expression includes the [`ANY` operator
+expression](/sql/functions/#expression-bool_op-any)
+(i.e., `ON fieldX = ANY(<array|list|map>)`), Materialize performs a cross join,
+which can lead to a significant increase in memory usage. If possible, rewrite
+the query to perform an equi-join on the unnested values.
+
+{{</ callout >}}
+
+## Idiomatic Materialize SQL
+
+**Idiomatic Materialize SQL:**   For equi-join whose `ON` expression includes
+the [`ANY` operator
+expression](/sql/functions/#expression-bool_op-any)
+(`ON fieldX = ANY(<array|list|map>)`), use [UNNEST()](/sql/functions/#unnest) in
+a [Common Table Expression (CTE)](/sql/select/#common-table-expressions-ctes) to
+unnest the values and perform the equi-join on the unnested values.
+
+<table>
+<thead>
+<tr>
+<th></th>
+<th></th>
+</tr>
+</thead>
+<tbody>
+<tr>
+<td><blue>Materialize SQL</blue></td>
+<td class="copyableCode">
+
+Use a Common Table Expression (CTE) to [UNNEST()](/sql/functions/#unnest) the
+array of values and perform the equi-join on the unnested values.
+
+<br>
+<div style="background-color: var(--code-block)">
+
+```mzsql
+WITH my_expanded_values AS
+(SELECT UNNEST(array_field) AS fieldZ FROM tableB)
```

Comment:
Is it worthwhile to call out that this only works if `array_field` does NOT contain duplicate entries. If it does, the query needs to be adapted to something slightly more complex. Something like this (untested):
```sql
WITH my_expanded_values AS
(SELECT UNNEST(array_field) AS fieldZ FROM tableB)
Comment
SELECT DISTINCT(<<all fields except the ones that gets joined over with any>>) a.fieldA, ...
FROM tableA a
JOIN my_expanded_values t ON a.fieldZ = t.fieldZ
ORDER BY <<all fields except the ones that gets joined over with any>>
```

Reply:
Sure.  Let me test it out and then add to the page.

Reply:
I ended up adding DISTINCT to the CTE's `UNNEST` projection `DISTINCT unnest(array_fields)`

Adding DISTINCT to the join's SELECT instead of the CTE means that if there are duplicate rows either tableA/tableB, we'd remove those duplicate results, which differs from the ANY sql results.  Now, the removal is probably what they want, but in that case, users would already have the DISTINCT in their original ANY sql projection, and thus, the rewrite doesn't change that.

Reply:
Good catch! It's much more natural this way. 

In practice, people likely need to expose additional columns from `tableB`, so it may make sense to change the query to something like
```sql
WITH my_expanded_values AS
(SELECT DISTINCT *, UNNEST(array_field) AS fieldZ FROM tableB)
```
but that's just a minor detail.

Reply:
I just noticed that you have a concrete example further below, which does that. So I don't think it's required to add it here!

Reply:
👍 -- if it turns out people do need it here as well, can easily pop it in.